pkg/kubelet/prober/prober.go

```diff
@@ -122,39 +116,31 @@ func (pb *prober) probe(ctx context.Context, probeType probeType, pod *v1.Pod, s
 
 // runProbeWithRetries tries to probe the container in a finite loop, it returns the last result
 // if it never succeeds.
-func (pb *prober) runProbeWithRetries(ctx context.Context, probeType probeType, p *v1.Probe, pod *v1.Pod, status v1.PodStatus, container v1.Container, containerID kubecontainer.ContainerID, retries int) (probe.Result, string, error) {
+func (pb *prober) runProbeWithRetries(ctx context.Context, p *v1.Probe, pod *v1.Pod, status v1.PodStatus, container v1.Container, containerID kubecontainer.ContainerID, req *http.Request, retries int) (probe.Result, string, error) {
 	var err error
 	var result probe.Result
 	var output string
 	for i := 0; i < retries; i++ {
-		result, output, err = pb.runProbe(ctx, probeType, p, pod, status, container, containerID)
+		result, output, err = pb.runProbe(ctx, p, pod, status, container, containerID, req)
 		if err == nil {
 			return result, output, nil
 		}
 	}
 	return result, output, err
 }
 
-func (pb *prober) runProbe(ctx context.Context, probeType probeType, p *v1.Probe, pod *v1.Pod, status v1.PodStatus, container v1.Container, containerID kubecontainer.ContainerID) (probe.Result, string, error) {
+func (pb *prober) runProbe(ctx context.Context, p *v1.Probe, pod *v1.Pod, status v1.PodStatus, container v1.Container, containerID kubecontainer.ContainerID, req *http.Request) (probe.Result, string, error) {
```

Comment:
the con I see with this approach is that we may be expanding the function signature for each probe type.
The same reasoning used for the httpprobe can be applied to the GRPC and TCP probes, no?

/cc @SergeyKanzhelev

Reply:
@aojea @SergeyKanzhelev I can create a structure `probeOptions` and pass that around ?

Reply:
creating structure that will handle different behaviors would be ideal